src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -34,30 +48,68 @@ public override Task<Unit> Handle(DidChangeTextDocumentParams request, Cancellat
             // we have full sync enabled, so apparently first change is the whole document
             var contents = request.ContentChanges.First().Text;
 
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, contents);
+            var documentUri = request.TextDocument.Uri;
+
+            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, contents);
+
+            if (IsBicepConfigFile(documentUri))
+            {
+                var configuration = configurationManager.GetConfiguration(documentUri.ToUri());
+                activeBicepConfigCache.AddOrUpdate(documentUri, (documentUri) => configuration, (documentUri, prevConfiguration) => configuration);
+            }
 
             return Unit.Task;
         }
 
         public override Task<Unit> Handle(DidOpenTextDocumentParams request, CancellationToken cancellationToken)
         {
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
+            var documentUri = request.TextDocument.Uri;
+
+            if (IsBicepConfigFile(documentUri))
+            {
+                var configuration = configurationManager.GetConfiguration(documentUri.ToUri());
+                activeBicepConfigCache.TryAdd(documentUri, configuration);
+            }
+
+            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
 
             return Unit.Task;
         }
 
         public override Task<Unit> Handle(DidSaveTextDocumentParams request, CancellationToken cancellationToken)
         {
+            var documentUri = request.TextDocument.Uri;
+
+            if (IsBicepConfigFile(documentUri) &&
+                activeBicepConfigCache.TryRemove(documentUri, out RootConfiguration? prevBicepConfiguration) &&
```

Comment:
Why are we removing it on save?

Reply:
On save, we want to send telemetry event. We can either do a TryGetValue(..) and then TryRemove(..) to keep the cache updated or TryRemove(..).

Let me know if I am missing something.

Reply:
Added some comments:
https://github.com/Azure/bicep/commit/1010ff5cc71629fccd7c55428ea3d10a167c03e1

Reply:
I'm still missing something.  Don't you still need to keep the cache around in case the file is changed/saved again?

Reply:
When the file is changed, the one from disk gets saved in cache. We don't use it until it's saved.

Reply:
Nope, still confused.  You save to the cache on open and GetTextDocumentAttributes, whatever that is.  You remove on close. All that makes sense.  But why are you removing on save instead of updating?

BTW, most of the comments you added aren't helpful, e.g.:
```
  // If the documentUri corresponds to bicepconfig.json and there's an entry in activeBicepConfigCache,
            // we'll use the last known configuration and the one from currently saved config file to figure out
            // if we need to send out telemetry information regarding the config change.
            // We'll also remove the entry from activeBicepConfigCache.
```
All of that is obvious from the code.  What's missing is why you're doing it.